### PR TITLE
Replace `msh` in public API comment based documentation with PowerShell equivalent

### LINKED ISSUE #12839: PowerShell is still sometimes called "Monad" or "MSH" in documentation
In some places, the PowerShell SDK documentation, and possibly other documentation, though I haven't checked thoroughly, still refers to "Monad" or "MSH," PowerShell's previous names during its original development. This has the potential to confuse programmers who are not aware of PowerShell's history.

As particular examples, this can be seen in [Cmdlet](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.cmdlet)'s and [RuntimeException](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.runtimeexception)'s SDK documentation, but it also occurs in many other documents.

As a documentation problem, this issue was originally opened as MicrosoftDocs/PowerShell-Docs#6043, but it was closed and reopened here because the documentation in question is generated from `///` doc-comments in the PowerShell codebase itself.

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

We are not generating and publishing API docs based on comment based docs and many still refer to MSH.  This PR updates those references to PowerShell equivalent.  The term monad is handled in a different PR.  Note non-public classes/methods aren't being changed in this PR.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/12839

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by JamesWTruher — APPROVED** [review]
LGTM

**Review by daxian-dbw — APPROVED** [review]
LGTM

File: src/System.Management.Automation/engine/cmdlet.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ namespace System.Management.Automation
     /// deriving from the PSCmdlet base class.  The Cmdlet base class is the primary means by
     /// which users create their own Cmdlets.  Extending this class provides support for the most
     /// common functionality, including object output and record processing.
-    /// If your Cmdlet requires access to the MSH Runtime (for example, variables in the session state,
+    /// If your Cmdlet requires access to the PowerSHell Runtime (for example, variables in the session state,
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/cmdlet.cs:26]:
```suggestion
    /// If your Cmdlet requires access to the PowerShell Runtime (for example, variables in the session state,
```

**Review by PaulHigin — CHANGES_REQUESTED** [review]

**Review by PaulHigin — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
The Linux test failure is a timing issue with the prediction input test and not related to these changes.  Subsequent re-runs is just wasting compute at this point.

**Comment by daxian-dbw** [comment]
You have to re-run the build stage in this case instead of the test stage. Re-runing now.

File: src/System.Management.Automation/engine/hostifaces/MshHost.cs [hunk 2]
```diff
@@ -64,9 +64,9 @@ protected PSHost()
         /// The name identifier of the hosting application.
         /// </value>
         /// <example>
-        ///     <MSH>
+        ///     <PowerShell>
         ///         if ($Host.Name -ieq "ConsoleHost") { write-host "I'm running in the Console Host" }
-        ///     </MSH>
+        ///     </PowerShell>
```
Comment (sdwheeler) [inline src/System.Management.Automation/engine/hostifaces/MshHost.cs:69]: `<PowerShell></PowerShell>` is not a valid XML element for ///-comment documentation.

**Pull request merged**